### PR TITLE
Docs: update frontend style guide to highlight new function paradigm

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR updates frontend style guide to highlight new function paradigm and adds section for default exports.

**Special notes for your reviewer**:

### CODE REVIEW

**Review by hugohaggmark — APPROVED** [review]
Left some comments but great work 🎉  We should wait for feedback from @achatterjee-grafana before merging

**Review by achatterjee-grafana — APPROVED** [review]
One nit, otherwise LGTM.

**Pull request merged**
